src/queue.ts

```diff
@@ -97,8 +97,8 @@ export class Queue<
   }
 
   /** Decode the payload, stripping away the outer JSON encoding */
-  static decodePayload<T>(s: string | null) {
-    return JSON.parse(s ?? "{}")._ as T;
+  static decodePayload<T>(s: string | null | unknown) {
+    return JSON.parse((s as string) ?? "{}")._ as T;
```

Comment:
This will probably need a try catch around it now, since non-strings really mess up the JSON.parse (it probably always needed one, but the addition of an `unknown` makes it way more likely to trigger this footgun)